File: healthcarePatient/src/com/Patient.java
package com;

import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.sql.Statement;

public class Patient {
	
	private Connection connect()
	{
			Connection con = null;
			try {
				
				Class.forName("com.mysql.jdbc.Driver");
				con = DriverManager.getConnection("jdbc:mysql://localhost:3306/helthcare","root","root");
				
			} catch (ClassNotFoundException e) {
				
				 e.printStackTrace();
				 
			}catch (SQLException e) {	
				
				 e.printStackTrace();
			}
			
		return con;
	}
	
	public String addPatient(String f_name, String l_name, String address, String dob, String phoneNo, String gender, String BloodGroup, String NIC)
	{
			String output = "";
			
			try
			{
				Connection con = connect();
	 
				if (con == null)
				{return "Error while connecting to the database for inserting."; }
	
				// create a prepared statement
				String sql = "INSERT INTO patient(f_name, l_name, address, dob, phoneNo, gender, BloodGroup, NIC) "+ "VALUE(?,?,?,?,?,?,?,?)";
	 
				PreparedStatement preparedStmt = con.prepareStatement(sql);
	 
				// binding values
				preparedStmt.setInt(1, 0);
				preparedStmt.setString(2, f_name);
				preparedStmt.setString(3, l_name);
				preparedStmt.setString(4, address);
				preparedStmt.setString(5, dob);
				preparedStmt.setInt(6, Integer.parseInt(phoneNo)); 
				preparedStmt.setString(7, gender);
				preparedStmt.setString(8, BloodGroup);
				preparedStmt.setString(9, NIC);
				
				preparedStmt.execute();
				con.close();
				
				String newPatient = ViewPatient(); 
				output = "{\"status\":\"success\", \"data\": \"" +newPatient + "\"}"; 
			}
			 catch (Exception e)
			 {
				 	output = "{\"status\":\"error\", \"data\":\"Error while inserting the Patient.\"}";
				 	System.err.println(e.getMessage());
			 }
			
			 return output;
			 
		}
			
	public String ViewPatient()

	{
	 String output = "";
	 
	 try
	 {
		 Connection con = connect();
		 if (con == null)
		 {
			 return "Error while connecting to the database for reading.";
		 }
		 // Prepare the html table to be displayed
		 output = "<table border='1'><tr><th>First Name</th><th>Last Name</th><th>Address</th><th>date of birth</th><th>contact</th><th>Gender</th><th>Blood Group</th><th>NIC</th><th>Update</th><th>Remove</th></tr>";
		 
		 String sql = "select * from patient";
		 Statement stmt = con.createStatement();
	     ResultSet rs = stmt.executeQuery(sql);
	 
	    // iterate through the rows in the result set
	    while (rs.next())
	    {
	    	String id = Integer.toString(rs.getInt("id"));
	    	String f_name = rs.getString("f_name");
	    	String l_name = rs.getString("l_name");
	    	String address = rs.getString("address");
	    	String dob = rs.getString("dob");
	    	String phoneNo = Integer.toString(rs.getInt("phoneNo"));
	    	String gender = rs.getString("gender");
	    	String BloodGroup = rs.getString("BloodGroup");
	    	String NIC = rs.getString("NIC");
	    	
	    	// Add into the html table
	    	output += "<tr><td><input id='hidPatientIDUpdate' name='hidPatientIDUpdate' type='hidden' value='" + id + "'>"+ f_name + "</td>";
	    	output += "<td>" + l_name + "</td>";
	    	output += "<td>" + address + "</td>";
	    	output += "<td>" + dob + "</td>";
	    	output += "<td>" + phoneNo + "</td>";
	    	output += "<td>" + gender + "</td>";
	    	output += "<td>" + BloodGroup + "</td>";
	    	output += "<td>" + NIC + "</td>";
	
	    	// buttons
	    	output +=  "<td><input name='btnUpdate' type='button'value='Update'class='btnUpdate btn btn-secondary'></td>"+"<td><input name='btnRemove' type='button' value='Remove'class='btnRemove btn btn-danger' data-id='"+ id + "'>" + "</td></tr>"; 
	    			
	    }
	    
	    con.close();
	    // Complete the html table
	    output += "</table>";
	 }
	 catch (Exception e)
	 {
		 output = "Error while reading the Patient.";
		 System.err.println(e.getMessage());
	 }
	return output; 
	}

	public String UpdatePatient(String id,String f_name, String l_name, String address, String dob, String phoneNo, String gender, String BloodGroup, String NIC)
	 {
					String output = "";
			
					try
					{
							Connection con = connect();
			 
							if (con == null)
							{return "Error while connecting to the database for updating."; }
			
							// create a prepared statement
							String sql = "UPDATE  patient SET " + "f_name = ?," + "l_name = ?,"+ "address = ?," + "dob = ?,"+ "phoneNo = ?," + "gender = ?," + "BloodGroup = ?," + "NIC = ?" + "WHERE id = ?";
			
							PreparedStatement preparedStmt = con.prepareStatement(sql);
			 
							// binding values
							preparedStmt.setString(1, f_name);
							preparedStmt.setString(2, l_name);
							preparedStmt.setString(3, address);
							preparedStmt.setString(4, dob);
							preparedStmt.setInt(5, Integer.parseInt (phoneNo)); 
							preparedStmt.setString(6, gender);
							preparedStmt.setString(7, BloodGroup);
							preparedStmt.setString(8, NIC);
							preparedStmt.setInt(9, Integer.parseInt(id));
			 
							// execute the statement
							preparedStmt.execute();
							con.close();
							
							String newPatient = ViewPatient();
							output =  "{\"status\":\"success\", \"data\": \"" +newPatient + "\"}"; 
			 }
			 catch (Exception e)
			 {
				 		output = "{\"status\":\"error\", \"data\":\"Error while updating the Patient.\"}";
				 		System.err.println(e.getMessage());
			 }
					
			 return output;
}

	public String deletePatient(String id)
	{
			String output = "";
	 
			try
			{
					Connection con = connect();
	 
					if (con == null)
					{return "Error while connecting to the database for deleting."; }
	 
					// create a prepared statement
					String sql = "delete from Patient where id=?";
	
					PreparedStatement preparedStmt = con.prepareStatement(sql);
	
					// binding values
					preparedStmt.setInt(1, Integer.parseInt(id));
	 
					// execute the statement
					preparedStmt.execute();
					con.close();
	 
					String newPatient = ViewPatient();
					output =  "{\"status\":\"success\", \"data\": \"" +newPatient + "\"}"; 
	 }
	 catch (Exception e)
	 {
		 		output =  "{\"status\":\"error\", \"data\":\"Error while deleting the Patient.\"}";
		 		System.err.println(e.getMessage());
	 }
	 return output;
	 } 
}

File: healthcarePatient/WebContent/Components/Patient.js
$(document).ready(function()
	{
		
		$("#alertSuccess").hide();     
		$("#alertError").hide();  
		
	});

//save----------
$(document).on("click", "#btnSave", function(event)
	{
		$("#alertSuccess").text("");
		$("#alertSuccess").hide();
		$("#alertError").text("");
		$("#alertError").hide();
	
		var status = validatePatientForm();
		
		//if not valid
		if (status != true)
		{
			$("#alertError").text(status);
			$("#alertError").show();
			return;
		}
	
		//if valid
		var type = ($("#hidPatientIDSave").val() == "") ? "POST" : "PUT";
		
		$.ajax(
				{
				 url : "PatientAPI",
				 type : type,
				 data : $("#formPatient").serialize(),
				 dataType : "text",
				 complete : function(response, status)
				 {
					 onPatientSaveComplete(response.responseText, status);
				 }
			});
	
	});


function onPatientSaveComplete(response, status)
{
		if (status == "success")
		{
				var resultSet = JSON.parse(response);
				
				if (resultSet.status.trim() == "success")
				{
					$("#alertSuccess").text("Successfully saved.");
					$("#alertSuccess").show();
					$("#divPatientGrid").html(resultSet.data);
 
				} else if (resultSet.status.trim() == "error")
				{
					$("#alertError").text(resultSet.data);
					$("#alertError").show();
				}
				
		} else if (status == "error"){
			
				$("#alertError").text("Error while saving.");
				$("#alertError").show();
			
		} else{
		
			$("#alertError").text("Unknown error while saving..");
			$("#alertError").show();
			}
				
			$("#hidPatientIDSave").val("");
			$("#formPatient")[0].reset();
}

function validatePatientForm()
{
	if ($("#f_Name").val().trim() == ""){
		return "Insert first Name.";	
	}
	
	// l_name
	if ($("#l_name").val().trim() == ""){
		return "Insert last Name.";	
	}
	
	// address-------------------------------
	if ($("#address").val().trim() == ""){
		return "Insert Address.";
		
	}
	
	// date of birth------------------------
	if ($("#dob").val().trim() == ""){
		
		return "Insert date of birth.";
	}
	
	// phoneNo-------------------------------
	if ($("#phoneNo").val().trim() == ""){
		return "Insert phoneNo.";
		
	}

	// gender-------------------------------
	if ($("#gender").val().trim() == ""){
		return "Insert gender.";
		
	}
	
	// BloodGroup-------------------------------
	if ($("#BloodGroup").val().trim() == ""){
		return "Insert Blood Group.";
		
	}
	
	// NIC-------------------------------
	if ($("#NIC").val().trim() == ""){
		return "Insert NIC.";
		
	}
	
	return true;
}

$(document).on("click", ".btnUpdate", function(event)
		{
			$("#hidPatientIDSave").val($(this).closest("tr").find('#hidPatientIDUpdate').val());
			$("#f_Name").val($(this).closest("tr").find('td:eq(0)').text());
			$("#l_name").val($(this).closest("tr").find('td:eq(1)').text());
			$("#address").val($(this).closest("tr").find('td:eq(2)').text());
			$("#dob").val($(this).closest("tr").find('td:eq(3)').text()); 
			$("#phoneNo").val($(this).closest("tr").find('td:eq(4)').text()); 
			$("#gender").val($(this).closest("tr").find('td:eq(5)').text()); 
			$("#BloodGroup").val($(this).closest("tr").find('td:eq(6)').text()); 
			$("#NIC").val($(this).closest("tr").find('td:eq(7)').text()); 
		});



$(document).on("click", ".btnRemove", function(event)
		{
		 $.ajax(
				 {
					 url : "PatientAPI",
					 type : "DELETE",
					 data : "id=" + $(this).data("id"),
					 dataType : "text",
					 complete : function(response, status)
					 {
						 onPatientDeleteComplete(response.responseText, status);
					 }
				 });
		});
function onPatientDeleteComplete(response, status)
{
		if (status == "success")
		{
			var resultSet = JSON.parse(response);
			
			if (resultSet.status.trim() == "success")
			{
				$("#alertSuccess").text("Successfully deleted.");
				$("#alertSuccess").show();
				$("#divPatientGrid").html(resultSet.data);
				
			} else if (resultSet.status.trim() == "error")
			{
				$("#alertError").text(resultSet.data);
				$("#alertError").show();
			}
		} else if (status == "error")
		{
			$("#alertError").text("Error while deleting.");
			$("#alertError").show();
		} else
		{
			$("#alertError").text("Unknown error while deleting..");
			$("#alertError").show();
		}
}




